ledger/eval.go

```diff
@@ -36,6 +36,11 @@ import (
 // ErrNoSpace indicates insufficient space for transaction in block
 var ErrNoSpace = errors.New("block does not have space for transaction")
 
+// maxPaysetHint makes sure that we don't allocate too much memory up front
+// in the block evaluator, since there cannot reasonably be more than this
+// many transactions in a block.
+const maxPaysetHint = 20000
```

Comment:
I'd be ok with changing it to a constant 15,000 since I think there can't be more than that many transactions in a block, but since `TxPoolSize` is a variable that may be changed in `config.json` to any value, I don't think it is appropriate to use it here -- `maxPaysetHint` should just be >= the maximum number of transactions that could ever appear in a block, and < some value that could cause problems due to using a lot of memory.

Reply:
`maxPaysetHint` is a hint; incorrectly setting it would not make anything harmful other than inefficient allocation.

I don't think that we need this to start with, as the hint should suffice ( and, in practice, it already provides a cap ).

When it comes from the eval path, the limit matching the number of transaction in the block.
When it comes from the TxPool, the limit is the number of transactions that are in the TxPool, which is already capped at `TxPoolSize`. ( i.e. the actual number is abit higher, as it also include dups )

Either way, I'm going to merge this as is. If you feel like changing this to TxPoolSize or some other number, I'll be happy to approve and merge it on a separate PR.